go/runtime/txpool/main_queue.go

```diff
@@ -107,17 +107,26 @@ func (mq *mainQueue) HandleTxsUsed(hashes []hash.Hash) {
 
 func (mq *mainQueue) GetSchedulingExtra(offset *hash.Hash, limit uint32) []*TxQueueMeta {
 	txMetas := mq.inner.getPrioritizedBatch(offset, limit)
-	var txs []*TxQueueMeta
+	txs := make([]*TxQueueMeta, 0, len(txMetas))
 	for _, txMeta := range txMetas {
 		txs = append(txs, &txMeta.TxQueueMeta) //nolint:gosec
 	}
 	return txs
 }
 
+func (mq *mainQueue) PeekAll() []*TxQueueMeta {
```

Comment:
```go
func (mq *mainQueue) GetTxsToPublish() []*TxQueueMeta {
	return mq.PeekAll()
}
```
To avoid code duplication. Maybe we can even get rid of `GetTxsToPublish`, did not check that.